src/par_iter/string.rs

```diff
@@ -68,3 +81,103 @@ impl<'a> IntoIterator for ParChars<'a> {
         self.chars.chars()
     }
 }
+
+
+// /////////////////////////////////////////////////////////////////////////
+
+pub struct ParSplit<'a> {
+    chars: &'a str,
+    separator: char,
+
+    /// Keeps track of the first separator found in the string.  This lets us
+    /// quickly answer `can_split`, and it also corresponds to what parts we've
+    /// already scanned as we keep splitting smaller.
+    first: Option<usize>,
+}
+
+impl<'a> ParSplit<'a> {
+    fn new(chars: &'a str, separator: char) -> Self {
+        ParSplit {
+            chars: chars,
+            separator: separator,
+            first: chars.find(separator),
+        }
+    }
+}
+
+impl<'a> ParallelIterator for ParSplit<'a> {
+    type Item = &'a str;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(self, consumer)
+    }
+}
+
+impl<'a> UnindexedProducer for ParSplit<'a> {
+    fn can_split(&self) -> bool {
+        self.first.is_some()
+    }
+
+    fn split(self) -> (Self, Self) {
+        let ParSplit { chars, separator, first } = self;
+        let first = first.expect("error: splitting with no separator present!");
+
+        // First find a suitable UTF-8 boundary in the unsearched region.
+        let char_index = find_char_midpoint(&chars[first..]) + first;
+
+        // Find a separator in reverse, towards the `first` that we know exists.
+        let index = chars[first..char_index]
+            .rfind(separator)
+            .map(|i| i + first)
+            .unwrap_or(first);
+
+        // Create the left side of the split.  It might not have a `first` anymore
+        // if that's the exact separator we're splitting on now.
+        let left_first = if first < index { Some(first) } else { None };
+        let left_split = ParSplit {
+            chars: &chars[..index],
+            separator: separator,
+            first: left_first,
+        };
+
+        // Create the right side of the split starting just after this separator.
+        // We find its `first` starting from the `char_index` already scanned above.
+        let right_index = index + separator.len_utf8();
+        let right_search = max(char_index, right_index);
+        let right_first = chars[right_search..]
+            .find(separator)
+            .map(|i| i + (right_search - right_index));
+        let right_split = ParSplit {
+            chars: &chars[right_index..],
+            separator: separator,
+            first: right_first,
+        };
+
+        // All done, now we are two!
+        (left_split, right_split)
+    }
+}
+
+impl<'a> IntoIterator for ParSplit<'a> {
+    type Item = &'a str;
+    type IntoIter = Chain<Once<&'a str>, Split<'a, char>>;
+
+    fn into_iter(self) -> Self::IntoIter {
```

Comment:
It is a good candidate to implement a custom `fold_with`, but we'll still have to have `IntoIterator` to satisfy that default implementation of `fold_with`.  (unless we cheat with `unimplemented!()`)